doctrine-mapping.xsd

```diff
@@ -424,7 +424,7 @@
 
   <xs:simpleType name="fqcn" id="fqcn">
     <xs:restriction base="xs:token">
-      <xs:pattern value="[a-zA-Z_u01-uff][a-zA-Z0-9_u01-uff]+" id="fqcn.pattern">
+      <xs:pattern value="([a-zA-Z_&#x7F;-&#x10FFFF;][a-zA-Z0-9_&#x7F;-&#x10FFFF;]*\\)*[a-zA-Z_&#x7F;-&#x10FFFF;][a-zA-Z0-9_&#x7F;-&#x10FFFF;]*" id="fqcn.pattern">
```

Comment:
Class names must follow `^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]+$` (https://github.com/Ocramius/ProxyManager/blob/99542a2c0fef18e0b460cbed93c5cde63cc2be91/tests/ProxyManagerTest/Generator/Util/UniqueIdentifierGeneratorTest.php#L42)

Reply:
First, they must follow `^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$` (with a `*`, not a `+`, on the character class for after the first char) according to http://php.net/manual/en/language.oop5.basic.php and verified at https://3v4l.org/klVjj.
=> Do you want to consider single-char names as invalid?

Second, that is a *mono-byte* regex, as explained at https://stackoverflow.com/a/5358944 (modulo the "*accidentally*") and verified at https://3v4l.org/XqJGf which shows that for UTF-8 encoded strings, the equivalent of
`/^[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*$/` is
`/^[a-zA-Z_\x7f-\x{10ffff}][a-zA-Z0-9_\x7f-\x{10ffff}]*$/u`.
Now, XML Regex is _Unicode mode_ (and all the files are in UTF-8 encoding), so I translated `\x7f-\x{10ffff}` to `&#x7F;-&#x10FFFF;`, because `&#x7F;-&#xFF;` (translated from `\x7f-\xff`) only matches Unicode characters in the range U+7F..U+FF, and I have indeed tested that it would then consider non-Latin1 characters as invalid (e.g. "The value 'Слон' is not accepted by the pattern ...").
=> Do you want to consider non-Latin1 (or even non-ASCII) names as invalid?

Reply:
Hmm, indeed about `*` (probably lucky that nobody hit the pattern before - the library always generates 2+ chars for that particular use-case).

As for the multi-byte, I'm inclined to say that supporting multibyte strings for class names is a no-go for a lot of stuff, because most of the `get_class()`-based operations (not just in ORM, but in other libraries) simply don't operate under those circumstances, and you will get someone doing a nice unsafe `substr()` somewhere.

I'd say that we should keep those out of support: that would be an opinionated stance (by me), not the "full support for everything".

Reply:
So the `&#x7F;-&#x10FFFF;` part should become just `&#x7F;` (because `&#x7F;-&#xFF;` would still accept e.g. the character `é` = `&#xE9;` which is 2-bytes `\xC3\xA9` in UTF-8), or even be removed entirely (do you accept the `DEL` char)?
And I should remove the test file `pattern-fqcn-valid-unicode.xml`, or alternatively rename it to `pattern-fqcn-invalid-multibyte.xml` and keep only the `"Éléphant"` case?